apm-sniffer/apm-agent/src/main/java/org/apache/skywalking/apm/agent/SkyWalkingAgent.java

```diff
@@ -75,12 +76,14 @@ public static void premain(String agentArgs, Instrumentation instrumentation) th
 
         new AgentBuilder.Default(byteBuddy)
             .ignore(nameStartsWith("net.bytebuddy."))
-            .ignore(nameStartsWith("org.slf4j."))
-            .ignore(nameStartsWith("org.apache.logging."))
-            .ignore(nameStartsWith("org.groovy."))
-            .ignore(nameContains("javassist"))
-            .ignore(nameContains(".asm."))
-            .ignore(allSkyWalkingAgentExcludeToolkit())
+                .or(nameStartsWith("org.slf4j."))
+                .or(nameStartsWith("org.apache.logging."))
+                .or(nameStartsWith("org.groovy."))
+                .or(nameContains("javassist"))
+                .or(nameContains(".asm."))
+                .or(nameStartsWith("sun.reflect"))
+                .or(allSkyWalkingAgentExcludeToolkit()
+                .or(ElementMatchers.<TypeDescription>isSynthetic()))
```

Comment:
Why need this? I thought `isSynthetic` should be field modifier? Which class definition is with this?

Reply:
For what I know the synthetic key word could also describe a class, that means compiler can generate synthetic class.
Exactly I am not sure what kinds of scene is this for, but consult to bytebuddy source code. It will create a default ignore when use the constructor `public Default(ByteBuddy byteBuddy) {...}` like this(L8334 in AgentBuilder.java):
```java
new RawMatcher.Disjunction(
                            new RawMatcher.ForElementMatchers(any(), isBootstrapClassLoader()),
                            new RawMatcher.ForElementMatchers(nameStartsWith("net.bytebuddy.").or(nameStartsWith("sun.reflect.")).<TypeDescription>or(isSynthetic()))),
```

Reply:
The `Class` class also have the `isSynthetic` modifier, what got from JDK source code.